File: stock-market-service/src/main/java/rs/edu/raf/repository/ListingOwnerRepository.java
package rs.edu.raf.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import rs.edu.raf.model.ListingOwner;

import java.util.Optional;

@Repository
public interface ListingOwnerRepository extends JpaRepository<ListingOwner, Long> {
    Optional<ListingOwner> findByOwnerAndTicker(Long owner, String ticker);
}

File: stock-market-service-kotlin/src/main/kotlin/org/example/stockmarketservicekotlin/repository/ListingOwnerRepository.kt
package org.example.stockmarketservicekotlin.repository

import org.example.stockmarketservicekotlin.model.ListingOwner
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface ListingOwnerRepository :JpaRepository<ListingOwner, Long>{
    fun findByOwnerAndTicker(owner: Long, ticker: String): Optional<ListingOwner>
}